base/acme/src/main/java/org/dogtagpki/acme/server/ACMEChallengeProcessor.java

```diff
@@ -121,6 +126,46 @@ public void finalizeInvalidAuthorization(Exception e) throws Exception {
         logger.info("Challenge " + challengeID + " is invalid");
         challenge.setStatus("invalid");
 
+        // RFC 8555 Section 7.1.6: Status Changes
+        //
+        // If the client attempts to fulfill a challenge and fails, or if there
+        // is an error while the authorization is still pending, then the
+        // authorization transitions to the "invalid" state.
+
+        logger.info("Authorization " + authzID + " is invalid");
+        authorization.setStatus("invalid");
+
         engine.updateAuthorization(account, authorization);
+
+        // RFC 8555 Section 7.1.6: Status Changes
+        //
+        // The order also moves to the "invalid" state if it expires or one of
+        // its authorizations enters a final state other than "valid" ("expired",
+        // "revoked", or "deactivated").
+
+        logger.info("Updating pending orders");
+
+        Collection<ACMEOrder> orders =
+            engine.getOrdersByAuthorizationAndStatus(account, authzID, "pending");
+
+        for (ACMEOrder order : orders) {
```

Comment:
Question: Should we be processing ALL orders or only the order that invoked this method?

Reply:
An authorization can be shared by multiple orders (although we haven't implemented it yet). When a client calls this method to validate a challenge, it doesn't specify which order it's using anyway, so we're updating all orders that depend on this authorization/challenge.

Reply:
FYI, we're doing the same thing in line 88 above.